core/src/main/java/org/jdbi/v3/core/extension/ExtensionContext.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.extension;
+
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+/**
+ * Accepted by a handle when executing extension code. This temporarily reconfigures the handle to execute
+ * extension code. Reconfigurations are e.g. annotations from the sql objects that define mappers or bindings.
+ */
+public final class ExtensionContext {
```

Comment:
I would have loved to just roll this into ExtensionMethod but it would change the constructor in a non-backwards compat way.

Reply:
If you add equivalent getters to this class, could we then deprecate `ExtensionMethod` and eventually retire it?

Reply:
there is a place where the config and the extension method show up on different call paths (the config comes from a lambda creation point and the extension method in the lambda execution), so I think for now we should keep them separate. We can't make it an internal class though because it is part of the extension API (HandleSupplier).